src/Compilers/Server/VBCSCompilerTests/CompilerServerHostTests.cs

```diff
@@ -0,0 +1,47 @@
+﻿using Roslyn.Test.Utilities;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+using static Microsoft.CodeAnalysis.CommandLine.BuildResponse;
+
+namespace Microsoft.CodeAnalysis.CompilerServer.UnitTests
+{
+    public sealed class CompilerServerHostTests
+    {
+        internal sealed class TestableCompilerServerHost : CompilerServerHost
+        {
+            public override IAnalyzerAssemblyLoader AnalyzerAssemblyLoader { get; }
+
+            public override Func<string, MetadataReferenceProperties, PortableExecutableReference> AssemblyReferenceProvider { get; }
+
+            public TestableCompilerServerHost(
+                IAnalyzerAssemblyLoader loader = null, 
+                Func<string, MetadataReferenceProperties, PortableExecutableReference> assemblyReferenceProvider = null) : 
+                base(ServerUtil.DefaultClientDirectory, ServerUtil.DefaultSdkDirectory)
+            {
+                AnalyzerAssemblyLoader = loader;
+                AssemblyReferenceProvider = assemblyReferenceProvider;
+            }
+
+            public override bool CheckAnalyzers(string baseDirectory, ImmutableArray<CommandLineAnalyzerReference> analyzers)
+            {
+                return true;
+            }
+        }
+
+        [WorkItem(13995, "https://github.com/dotnet/roslyn/issues/13995")]
+        [Fact]
+        public void RejectEmptyTempPath()
+        {
+            var host = new TestableCompilerServerHost();
+            var request = new RunRequest(LanguageNames.CSharp, currentDirectory: @"c:\current", tempDirectory: null, libDirectory: null, arguments: Array.Empty<string>());
```

Comment:
Consider using `currentDirectory: Temp.CreateDirectory().Path` instead, to ensure the path matches the current platform, and to ensure that if `RunRequest` actually writes to that path (now or in the future), the path is temporary.